File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> hash;
        int longest = 0;
        int curr = 0;
        
        for(int i=0; i<nums.size(); i++) hash.insert(nums[i]);
        
        int i=0;
        
        while(i<nums.size()){
            if(hash.find(nums[i] - 1) != hash.end()){
                i++;
            }else{
                curr++;
                int k = nums[i]+1;
                while(hash.find(k) != hash.end()) {
                    curr++;
                    k++;
                }
                longest = max(curr, longest);
                curr = 0;
                i++;
            }
        }
        
        return longest;
        
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        maxLen = 0
        nums = set(nums)
        for num in nums:
            if num-1 not in nums:
                currLen = 1
                
                while num + currLen in nums:
                    currLen += 1
                maxLen = max(currLen, maxLen)
            
        return maxLen